auth.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"bufio"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"golang.org/x/crypto/ssh/terminal"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"syscall"
+)
+
+const (
+	https        = "https"
+	accessToken = "access_token"
+)
+
+type tokenRequest struct {
+	User     string
+	Password string
+	Realm    *string
```

Comment:
The realm can be optional in general, so I though a nullable field would make here more sense.

Yet still I am considering whether it's going to be useful at all and whether it should not be removed at this stage and only added if there would be a real need for it.